src/routes/WalletRequiredRoute.tsx

```diff
@@ -50,13 +53,22 @@ const WalletRequiredRoute = ({
   const locationPath = location && `${location.pathname}${location.search}`;
   return (
     <Route
+      path={path}
       render={(
         props: ReactRouterComponentProps<
-          {},
+          { colonyName?: string },
           StaticContext,
-          { redirectTo?: string; colonyURL?: string }
+          {
+            redirectTo?: string;
+            colonyURL?: string;
+            isMetaColonyExclusive?: boolean;
+          }
         >,
       ) => {
+        const isMetaColonyExclusive =
+          props.match.path === CLAIM_TOKEN_ROUTE ||
+          props.match.path === UNWRAP_TOKEN_ROUTE;
+
```

Comment:
I like this, but as I mentioned above, I think we need to switch this to use the other type of route, meaning this won't be needed anymore

Reply:
Does this mean that we want to make these routes _not_ metacolony exclusive so that users are not redirected to a 404 page if they try to access the route from another colony?

Reply:
Yes